File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/inbound/HospitalService.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.inbound;

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*;
import java.math.*;
import java.time.*;
import java.util.*;

/**
 * Inbound Service Port for managing [Hospital, Doctor].
 */
@org.springframework.modulith.NamedInterface("HospitalService")
public interface HospitalService {

    /**
     *
     *
     */
    public Optional<Hospital> getHospital(Long id);
    /**
     *
     *
     */
    public Hospital createHospital(Hospital input);
    /**
     *
     *
     */
    public Optional<Hospital> updateHospital(Long id, Hospital input);
    /**
     *
     *
     */
    public List<Hospital> listHospitals();
    /**
     *
     *
     * With Events: [DoctorCreated].
     *
     */
    public Doctor createDoctor(Doctor input);
    /**
     *
     *
     */
    public Optional<Doctor> updateDoctor(Long id, Doctor input);
    /**
     *
     *
     */
    public Optional<Doctor> getDoctor(Long id);
    /**
     *
     *
     */
    public List<Doctor> listDoctors();
    /**
     *
     *
     */
    public List<Doctor> listHospitalDoctors(Long hospitalId);
    /**
     *
     *
     */
    public List<PatientHospital> listHospitalPatients(Long hospitalId);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/inbound/HospitalService.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.inbound

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*
import java.math.*
import java.time.*

/** Inbound Service Port for managing [Hospital, Doctor]. */
@org.springframework.modulith.NamedInterface("HospitalService")
interface HospitalService {

    /**  */
    fun getHospital(id: Long): Hospital?

    /**  */
    fun createHospital(input: Hospital): Hospital

    /**  */
    fun updateHospital(id: Long, input: Hospital): Hospital?

    /**  */
    fun listHospitals(): List<Hospital>

    /** With Events: [DoctorCreated]. */
    fun createDoctor(input: Doctor): Doctor

    /**  */
    fun updateDoctor(id: Long, input: Doctor): Doctor?

    /**  */
    fun getDoctor(id: Long): Doctor?

    /**  */
    fun listDoctors(): List<Doctor>

    /**  */
    fun listHospitalDoctors(hospitalId: Long): List<Doctor>

    /**  */
    fun listHospitalPatients(hospitalId: Long): List<PatientHospital>
}
